sc-kpm/sc-agents-common/agents/GenerationByTemplateAgent.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2022 Intelligent Semantic Systems LLC, All rights reserved.
+ */
+
+#include "sc-memory/scs/scs_types.hpp"
+
+#include "utils/CommonUtils.hpp"
+#include "utils/IteratorUtils.hpp"
+
+#include "GenerationByTemplateAgent.hpp"
+
+namespace scAgentsCommon
+{
+SC_AGENT_IMPLEMENTATION(GenerationByTemplateAgent)
+{
+  ScAddr questionNode = m_memoryCtx.GetEdgeTarget(edgeAddr);
+  if (!checkActionClass(questionNode))
+    return SC_RESULT_OK;
+
+  SC_LOG_DEBUG("GenerationByTemplateAgent started")
+
+  ScAddr templateLinkAddr = utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, questionNode, CoreKeynodes::rrel_1);
+  ScAddr parameterNodeAddr =
+      utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, questionNode, CoreKeynodes::rrel_2);
+
+  if (!templateLinkAddr.IsValid())
+  {
+    SC_LOG_ERROR("GenerationByTemplateAgent: template link not found.")
+    utils::AgentUtils::finishAgentWork(&m_memoryCtx, questionNode, false);
+    return SC_RESULT_ERROR;
+  }
+  if (!parameterNodeAddr.IsValid())
+  {
+    SC_LOG_ERROR("GenerationByTemplateAgent: parameter node not found.")
+    utils::AgentUtils::finishAgentWork(&m_memoryCtx, questionNode, false);
+    return SC_RESULT_ERROR;
+  }
+
+  std::string linkContent = utils::CommonUtils::getLinkContent(&m_memoryCtx, templateLinkAddr);
+
+  ScAddr ruleStatement = m_memoryCtx.HelperFindBySystemIdtf(linkContent);
```

Comment:
we need to do this only if the identifier string is not empty, right? maybe it's worth to add a check